Objects/unicodeobject.c

```diff
@@ -15657,65 +15657,57 @@ PyUnicode_InternFromString(const char *cp)
 }
 
 
-#if defined(WITH_VALGRIND) || defined(__INSURE__)
 static void
 unicode_release_interned(void)
 {
+    Py_ssize_t pos = 0;
+    PyObject *key, *value;
+
     if (interned == NULL || !PyDict_Check(interned)) {
         return;
     }
-    PyObject *keys = PyDict_Keys(interned);
-    if (keys == NULL || !PyList_Check(keys)) {
-        PyErr_Clear();
-        return;
-    }
 
     /* Since unicode_release_interned() is intended to help a leak
        detector, interned unicode strings are not forcibly deallocated;
        rather, we give them their stolen references back, and then clear
        and DECREF the interned dict. */
-
-    Py_ssize_t n = PyList_GET_SIZE(keys);
 #ifdef INTERNED_STATS
-    fprintf(stderr, "releasing %zd interned strings\n", n);
+    fprintf(stderr, "releasing %zd interned strings\n", PyDict_Size(interned));
 
     Py_ssize_t immortal_size = 0, mortal_size = 0;
 #endif
-    for (Py_ssize_t i = 0; i < n; i++) {
-        PyObject *s = PyList_GET_ITEM(keys, i);
-        if (PyUnicode_READY(s) == -1) {
+    while (PyDict_Next(interned, &pos, &key, &value)) {
```

Comment:
Please rename key to s: there is no need to change the current code. I also suggest to rename value as "ignored_value".

Reply:
got it.